### PR TITLE
Added published date as editable in admin articles list/show

### LINKED ISSUE #10871: Update post publish date as a Forem Admin
**Is your feature request related to a problem? Please describe.**

As a Forem Admin, I may sometimes need to manually adjust the publish date to account.  This is currently possible via `/resource_admin`, but not `/admin`. 

**Describe the solution you'd like**

As a Forem Admin with relevant permissions, I should be able to go into `/admin/articles` and manually adjust a publish date.

### PR DESCRIPTION
**Pull request by akashdotsrivastava**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds post `published_at` as an editable datetime select option in the admin side articles index/show views

## Related Tickets & Documents
Closes #10871 

## QA Instructions, Screenshots, Recordings

#QA
Check out a published article in admin/articles list/show. Update the published_at date to something else and hit `Submit`. Confirm it has changed

Screenshot 2020-10-27 at 10 45 10 PM

#Additional changes
1. Because `published_at` is added as a rails datetime_select field to fulfil all kinds of datetime requirements, the #update logic in admin/articles controller had to be changed. The existing logic picked up individual values from permitted params, changing a few of them, before saving them But datetime select pushes params as `param(1i)`, `param(2i)` and so on, which doesn't sit well with extracting params. It is custom made to work with `model.create(permitted_params)` or `model.update(permitted_params)`. Now, the code directly runs `article.update(article_params)` for checking successful update
2. Because params are not cleaned up anymore, had to make sure all checkboxes return the correct boolean from the form. Converted all checkboxes to `rails form check_box fields` to ensure right matching to form object.
3. Only co author ids needs special treatment for conversion to array from comma joined string before saving. To ensure smooth input from the form without changing the UI, added a model attribute `co_author_ids_list` which acts as a form field, and then parses it back to array at model level instead of controller, before saving

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] Docs.forem.com
- [ ] README
- [x] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by akashdotsrivastava** [comment]
Updated the PR with the latest master and added a request spec for the changes made

File: app/models/article.rb [hunk 1]
```diff
@@ -379,6 +379,10 @@ def update_score
                    spaminess_rating: BlackBox.calculate_spaminess(self))
   end
 
+  def co_author_ids_list=(list_of_co_author_ids)
+    self.co_author_ids = list_of_co_author_ids.split(",").map(&:strip)
+  end
```
Comment (juliannatetreault) [inline app/models/article.rb:384]: Great refactor!

**Review by juliannatetreault — APPROVED** [review]
Thanks so much for tackling this issue @akashdotsrivastava! I tested this out locally and can confirm that it works great and that the refactors made work as expected, too -- great job. The test that is failing on your Travis build looks like a known, flaky spec, so I'm going to try rerunning the build for you. Awesome contribution! 🚀

**Comment by akashdotsrivastava** [comment]
@jacobherrington This PR requires another review from you for merge. Just a reminder. Thanks 🙂

**Review by jacobherrington — APPROVED** [review]
Good refactor! Thanks for adding this :smile:

**Pull request merged**